### ISSUE TITLE
feat: on ion-router change events, attach information about route that will/did change

### ISSUE THREAD

**Issue by adrm** [ionic-team/ionic-framework#23026]
# Feature Request

**Ionic version:**
[x] **5.x**

**Describe the Feature Request**

Right now, there is no way for components being routed by ion-router to know if they are being shown. With ionic react/vue/angular this is not a problem because ionic is calling router lifecycle methods in the routed components, but I am using ion-router in an Stencil app, so there is no lifecycle method calling and the ionRouterWillChange and ionRouterDidChange events do not give any information about what component are we changing to.

This is a problem because routes are kept in the dom and returning to an already loaded route won't give any signal to the component that it is being shown so it can reset itself to some specific state. Subscribing to the ion-router's ionRouterWillChange event is of no use, because every already loaded component will receive the event but there is no way to know which of them is being loaded.

The same problem happens for ion-nav change events, as those also don't carry any info about what component is being affected by the navigation change.

**Describe Preferred Solution**

Ideally, ionRouterWillChange and ionRouterDidChange would carry some information about the route, so components can subscribe to this events and filter if the change affects them or not.

**Describe Alternatives**

Maybe I can create a custom FrameworkDelegate I can pass to ion-nav that would handle this and call a predefined method in my components when the page is about to change, but I think that is too complicated for this small need.

Right now I am subscribing to ionNavDidChange, requesting ionNav.getActive() to get the active page and comparing with the DOM element of every routed component to see if they are being navigated to. This is not ideal because I cannot start updating the component until the navigation is already done, since there is no way to get the page we are going to navigate to when ionNavWillChange is dispatched.

**Comment by liamdebeasi** [comment]
Thanks for the report. There has not been much support for this from the community since the issue was opened. Due to this, the team does not have plans to implement this in the immediate future. I am going to close this for now. However, we may re-consider this feature in the future if community support grows.
